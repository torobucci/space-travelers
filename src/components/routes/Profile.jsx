import React from 'react';
import { useSelector } from 'react-redux';
import Container from 'react-bootstrap/esm/Container';
import ListGroup from 'react-bootstrap/ListGroup';

const Profile = () => {
  const { missions } = useSelector((state) => state.missions);
  const { rockets } = useSelector((state) => state.rockets);
  const myMissions = missions.filter((mission) => mission.reserved);
  const myRockets = rockets.filter((mission) => mission.reserved);
  return (
    <Container className="d-flex">
      <Container>
        <h2>My Missions</h2>
        <ListGroup>
          {myMissions.map((mission) => (
            <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
          ))}
        </ListGroup>
      </Container>
      <Container>
        <h2>My Rockets</h2>
        <ListGroup>
          {myRockets.map((rocket) => (
            <ListGroup.Item key={rocket.id}>{rocket.name}</ListGroup.Item>
          ))}
        </ListGroup>
      </Container>
    </Container>

  );
};

export default Profile;
