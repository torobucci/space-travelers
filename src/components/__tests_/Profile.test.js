import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Profile from '../routes/Profile';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'

jest.mock('react-bootstrap/esm/Container', () => 'MockedBootstrapContainer');
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Profile component', () => {
  beforeEach(() => {
    useSelector.mockImplementation((callback) => callback({
      missions: {
        missions: [
          {
            mission_id: 1,
            mission_name: 'Mission 1',
            reserved: true,
          },
          {
            mission_id: 2,
            mission_name: 'Mission 2',
            reserved: false,
          },
        ],
      },
      rockets: {
        rockets: [
          {
            id: 1,
            name: 'Rocket 1',
            reserved: true,
          },
          {
            id: 2,
            name: 'Rocket 2',
            reserved: false,
          },
        ],
      },
    }));
  });

  afterEach(() => {
    useSelector.mockClear();
  });

  it('renders the My Missions and My Rockets sections', () => {
    const { getByText } = render(<Profile />);

    expect(getByText('My Missions')).toBeInTheDocument();
    expect(getByText('My Rockets')).toBeInTheDocument();
  });

  it('renders the reserved missions', () => {
    const { getByText, queryByText } = render(<Profile />);

    expect(getByText('Mission 1')).toBeInTheDocument();
    expect(queryByText('Mission 2')).not.toBeInTheDocument();
  });

  it('renders the reserved rockets', () => {
    const { getByText, queryByText } = render(<Profile />);

    expect(getByText('Rocket 1')).toBeInTheDocument();
    expect(queryByText('Rocket 2')).not.toBeInTheDocument();
  });
  it('Profile snapshots match', () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
