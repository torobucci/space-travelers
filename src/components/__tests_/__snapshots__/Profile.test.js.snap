// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Profile snapshots match 1`] = `
<MockedBootstrapContainer
  className="d-flex"
>
  <MockedBootstrapContainer>
    <h2>
      My Missions
    </h2>
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        Mission 1
      </div>
    </div>
  </MockedBootstrapContainer>
  <MockedBootstrapContainer>
    <h2>
      My Rockets
    </h2>
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        Rocket 1
      </div>
    </div>
  </MockedBootstrapContainer>
</MockedBootstrapContainer>
`;
